### PR TITLE
Keep focus when clicking inside Magic Search.

### LINKED ISSUE #12229: Themes: when clicking "×" in the search box shouldn't lose focus
cap-

Currently if `×` is pressed to clear the field in themes magic search the focus is lost.

The focus should stay on the field until the user taps outside.

### PR DESCRIPTION
**Pull request by budzanowski**
### Information 
This feature was requested in #12229. Additionally to the request from Issue this PR also extends requested behavior and now clicking on tier also preserves focus. 

Code in this PR may be seen as a bit controversial. the `X` in `Search` was disabled( standard prop no change required ) and MagicSearch implements it's own `X` functionality. The decision to go this was made because `X` functionality from search component was design and created as **close** and changing that would require deep modifications in `Search` 

fixes #12229 

### Testing
To verify that this works compare to the situation from #12229

### CODE REVIEW

**Comment by budzanowski** [comment]
[Themes: when clicking "×" in the search box shouldn't lose focus](https://trello.com/c/G8F7WHH2/1331-themes-when-clicking-in-the-search-box-shouldn-t-lose-focus)

File: client/my-sites/themes/themes-magic-search-card/index.jsx [hunk 1]
```diff
@@ -238,8 +252,23 @@ class ThemesMagicSearchCard extends React.Component {
...
 					{ searchField }
+					{ ! isMobile() && this.state.searchInput !== '' &&
+						<div className="themes-magic-search-card__icon" >
+							<Gridicon
+								icon="cross"
+								className="themes-magic-search-card__icon-close"
+								tabIndex="0"
+								onClick={ this.clearSearch }
+								aria-controls={ 'search-component-magic-search' }
+								aria-label={ translate( 'Clear Search', { context: 'button label' } ) }
```
Comment (a8ci18n) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Close Search', { context: 'button label'} )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

Comment (budzanowski) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: :-1:

File: client/my-sites/themes/themes-magic-search-card/style.scss [hunk 2]
```diff
@@ -7,7 +7,7 @@
 	align-items: center;
 	background: white;
 	box-shadow: 0 0 0 1px transparentize( lighten( $gray, 20% ), .5 ), 0 1px 2px lighten( $gray, 30% );
-	transition: all 0.15s ease-in-out;
+	//transition: all 0.15s ease-in-out;
```
Comment (folletto) [inline client/my-sites/themes/themes-magic-search-card/style.scss]: Why is this commented? Was there any issue with this transition?

**Comment by folletto** [comment]
Tested in Chrome, design wise 👍

File: client/my-sites/themes/themes-magic-search-card/style.scss (hunk 2 above)
Comment (budzanowski) [inline client/my-sites/themes/themes-magic-search-card/style.scss]: :( good catch it should not be commented out. Adding it back.

**Comment by ockham** [comment]
> The decision to go this was made because `X` functionality from search component was design and created as **close** and changing that would require deep modifications in `Search` 

Just wondering however -- do we consider the same behavior to be a bug in [`Search`](https://wpcalypso.wordpress.com/devdocs/design/search), too? (If so, we should fix it there...) cc @folletto

**Comment by folletto** [comment]
It's different for MagicSearch as it provides suggestions, while the normal search doesn't.
Also, the normal search component has `×` as a way to close the component, not just clear, which makes it even more different (and might still be the case with Magic Search's component too if the Search is a slide out instead of being default open).

tl;dr: no further change at this point.

File: client/my-sites/themes/themes-magic-search-card/index.jsx [hunk 3]
```diff
@@ -192,6 +189,24 @@ class ThemesMagicSearchCard extends React.Component {
...
+	clearSearch = () => {
+		this.updateInput( '' );
+		this.focusOnInput();
+	}
+
+	handleClickOutside() {
+		this.setState( { searchIsOpen: false } );
+	}
+
+	// TODO remember cursor position and put it back after focus
```
Comment (seear) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: > // TODO remember cursor position and put it back after focus

Not sure we would ever get around to this. And what about if user clicks on specific position on purpose, only for the cursor to jump to remembered position?

Maybe remove this comment.

File: client/my-sites/themes/themes-magic-search-card/index.jsx (hunk 1 above)
Comment (seear) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: > You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

This is correct, the `{ context: 'button label' }` part can be removed here.

**Comment by budzanowski** [comment]
Rebased.
